virtcontainers/device/api/interface.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2017-2018 Intel Corporation
+// Copyright (c) 2018 Huawei Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package api
+
+import (
+	"github.com/sirupsen/logrus"
+
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+)
+
+var devLogger = logrus.FieldLogger(logrus.New())
+
+// SetLogger sets the logger for virtcontainers package.
+func SetLogger(logger logrus.FieldLogger) {
+	devLogger = logger
+}
+
+// DeviceLogger returns logger for device management
+func DeviceLogger() *logrus.Entry {
+	return devLogger.WithField("subsystem", "device")
+}
+
+// DeviceReceiver is an interface used for accepting devices
+// a device should be attached/added/plugged to a DeviceReceiver
+type DeviceReceiver interface {
+	HotplugAddDevice(Device, config.DeviceType) error
+	HotplugRemoveDevice(Device, config.DeviceType) error
+
+	// this is only for virtio-blk support
```

Comment:
This is really only needed for virtio-scsi now that we have merged @267. And hopefully we can remove this in future by refactoring the scsi lun management code.

Reply:
I agree with you. These two interfaces are truly not good being here. 
This part would need more refactor as you said, I am trying to make this PR a little bit simplier so that it won't be too hard to review.